scalableminds/webknossos
# In proofreading, split a segment from all its neighbors

This PR implements a new "Split from all neighboring segments" feature for the proofreading mode. The front-end code regarding the proofreading was also refactored quite a bit to generalize a bit better (the new feature only requires one agglomerate ID as input instead of two and in contrast to the other split actions, the result can consist of more than 2 new agglomerates).

### URL of deployed dev instance (used for testing):
- https://splitfromall.webknossos.xyz

### Steps to test:
- create a new annotation for a DS with an agglomerate file
- enable a hdf5 mapping
- use proofreading tool
  - without skeletons:
    - rightclick a segment and select "Split from all neighboring segments" --> segment should be split as expected
  - with skeletons
    - load an agglomerate tree
    - rightclick a node and select "Split from all neighboring segments" --> segment should be split as expected
- also test the other proofreading tools (merge, cut, min-cut)

### TODOs:
  - [x] Add neighbors route `POST /tracings/volume/<id>/agglomerateGraphNeighbors`
    - Example body `{"segmentPosition":[100,119,95],"mag":[1,1,1],"agglomerateId":1,"editableMappingId":"15d8d14c-1145-43f8-80a8-9a1a065facc7"}`
    - Returns `{"segmentId":3,"neighbors":[{"segmentId":1,"position":[77,77,39]},{"segmentId":49,"position":[87,100,70]},{"segmentId":56,"position":[80,149,94]},{"segmentId":58,"position":[91,107,76]},{"segmentId":63,"position":[83,64,59]},{"segmentId":76,"position":[105,101,62]},{"segmentId":82,"position":[96,89,66]},{"segmentId":87,"position":[100,50,62]},{"segmentId":88,"position":[104,53,51]},{"segmentId":101,"position":[110,50,59]},{"segmentId":173,"position":[146,148,82]}]}`
  - [x] User interaction (without skeletons)
    - [x] Call the new api route, construct relevant (directed) edges from result for split actions, push those
  - [x] User interaction from skeleton

### Issues:
- fixes #7607

------
